docker_cli/_cli.py

```diff
@@ -0,0 +1,327 @@
+import argparse
+import sys
+import textwrap
+from typing import Sequence
+
+from ._utils import universal_tag_prefix
+from .commands import dropin, make_lockfile, remove
+from .setup_commands import (setup_all, setup_cuda_dev, setup_cuda_runtime,
+                             setup_env_add, setup_env_dev, setup_env_runtime,
+                             setup_init)
+
+
+def setup_parser() -> argparse.ArgumentParser:
+    """
+    Create a top-level argument parser.
+
+    Returns
+    -------
+    argparse.ArgumentParser
+        The parser.
+    """
+    prefix = universal_tag_prefix()
+
+    def help_formatter(prog): return argparse.HelpFormatter(
+        prog,
+        max_help_position=60
+    )
+
+    parser = argparse.ArgumentParser(
+        prog=__package__,
+        formatter_class=help_formatter
+    )
+
+    # Additional parsers for shared commands
+    setup_parse = argparse.ArgumentParser(add_help=False)
+    setup_parse.add_argument(
+        "--base", "-b", type=str, required=True,
+        help='The name of the base docker image.'
+    )
+
+    no_cache_parse = argparse.ArgumentParser(add_help=False)
+    no_cache_parse.add_argument(
+        "--no-cache",
+        action="store_true",
+        help="Run docker build with no cache if used."
+    )
+
+    cuda_run_parse = argparse.ArgumentParser(add_help=False)
+    cuda_run_parse.add_argument(
+        "--cuda-version", "-c", default="11.4", type=str,
+        help='The CUDA version. Default: "11.4"',
+        metavar="VERSION"
+    )
+    cuda_run_parse.add_argument(
+        "--cuda-repo", default="rhel8", type=str,
+        help="The name of the CUDA repository for this distro "
+        '(e.g. "rhel8", "ubuntu2004".) Default: "rhel8"',
+        metavar="REPO_NAME"
+    )
+
+    env_parse = argparse.ArgumentParser(add_help=False)
+    env_parse.add_argument(
+        "--env-file", default="spec-file.txt", type=str,
+        help="The location of the spec-file. Default: \"spec-file.txt\""
```

Comment:
My presumption has been that, for the typical case where the user just wants to build ISCE3 with a standard lockfile, there would be a file somewhere in this system that is always in the same place, and this default value would eventually point to that.

Reply:
Currently, ISCE3 has two lockfiles (and two requirements files) one for runtime dependencies and one for dev (build-time) dependencies. They're located inside the ISCE3 codebase, so they're locations *would* be fixed, but it's not clear which one of the two you would use as the default for this `--env-file` parameter since it's used in both the "runtime" and "dev" setup steps.

So since this `--env-file` parameter seems to be intended to be generic, I was thinking that we should remove the default value and make it required.

Alternatively, you could split it into two different parameters (one for runtime and one for dev) and give each one a different default value. Maybe that approach makes more sense. Then it could be a bit more consistent with the behavior of the `setup all` subcommand, which has a different default value for the runtime env file and the dev env file.

Reply:
Just read the comment below about the similar arguments under `setup_all_parser` - is there a reason why this part should be done differently? I could pretty easily remove env_parse and instead just have `setup conda dev` and `setup conda runtime` have their own `--env-file` arguments with the same defaults as their respective argument under `setup all`.

Reply:
>Just read the comment below about the similar arguments under `setup_all_parser` - is there a reason why this part should be done differently? I could pretty easily remove env_parse and instead just have `setup conda dev` and `setup conda runtime` have their own `--env-file` arguments with the same defaults as their respective argument under `setup all`.

Yeah, that sounds like a better approach than removing the default :+1:

Reply:
To this end, is there a naming scheme and location for these files which makes sense as a good default?

Reply:
Added a commit that puts two lockfiles in the `env_files` directory at the root of the project, called `lock-dev.txt` and `lock-runtime.txt`. The code now sets the defaults to look for them in this location relative to the module rather than the working directory. Let me know if there are any other names or locations you'd like for me to use.